src/frontend/src/app/assets/assets.component.ts

```diff
@@ -1,51 +1,54 @@
 import { Component, OnInit } from '@angular/core';
-import { AssetsService } from '../services/index';
+import { AssetsService, AlertService } from '../services/index';
 
 @Component({
   selector: 'app-assets',
   templateUrl: './assets.component.html',
   styleUrls: ['./assets.component.css']
 })
 export class AssetsComponent implements OnInit {
- assets = [];
- assetsReadingsData = [];
-  constructor(private assetService: AssetsService) { }
+  assets = [];
+  assetsReadingsData = [];
+  constructor(private assetService: AssetsService, private alertService: AlertService) { }
 
   ngOnInit() {
     this.getAsset();
   }
 
- public getAsset():void {
+  public getAsset(): void {
     this.assets = [];
     this.assetService.getAsset().
       subscribe(
       data => {
+        if (data.error) {
+          this.alertService.error(data.error.message)
```

Comment:
we should log the error in console too, what if alert service fails?

Reply:
Okay